content/blog/client_node_and_cncf_mentorship.md

```diff
@@ -0,0 +1,208 @@
+
+## Entering Neon
+
+[Neon](https://github.com/neon-bindings/neon) is an open-source Rust bindings for Nodejs. It allows me to use JavaScript types within Rust. Here is a simple Neon program
+
+```rust
+use neon::prelude::*;
+
+fn hello(mut cx: FunctionContext) -> JsResult<JsString> {
+    Ok(cx.string("hello node"))
+}
+
+#[neon::main]
+fn main(mut cx: ModuleContext) -> NeonResult<()> {
+    cx.export_function("hello", hello)?;
+    Ok(())
+}
+```
+
+This program exposes a Rust function named "hello", which takes a `FunctionContext` and returns a `JsResult<JsString>`. The function context is a control stack that manages memory and throws exception if needed. It also protects a pointer to a JavaScript value that is managed by JavaScript's garbage collector, making sure that the value's lifetime is valid during Rust's interaction with it. In the above example, `cx.string("hello node")` is a pointer (aka Handle) to an owned JavaScript string, represented as `JsString` in Rust. 
+
+After the program is compiled to a exeuctable "index.node", we can load it in nodejs runtime and call function "hello" like below: 
+
+```
+$ npm install
+$ node
+> require('.').hello()
+"hello node"
+```
+
+This is really nice, right? You can learn more about Neon in their official [documentation](https://neon-bindings.com/docs/hello-world). They wrote a few [examples](https://github.com/neon-bindings/examples) that I personally think very helpful. I also found that their Slack community is extremely friendly and welcoming.
+
+## Repo structure
+
+I first started writing Rust wrappers on Raw client - a low level API for querying/updating individual key-value pairs. The NodeJS client has a dependency on Rust client because that's one of the benefits using Neon - to reuse the code in Rust client. 
+
+The `client-node` repo structure looks like this:
+
+```rust
+├── Cargo.lock
+├── Cargo.toml
+├── index.node
+├── package.json
+├── src
+    └── lib.rs
+		... 
+├── tikv_client
+		└── index.js
+```
+
+The `[lib.rs](http://lib.rs)` implements the main logic that exposes the client functionalities. In each client method, the program parses all function arguments and pass them to native rust-client function. Then returns the result. Here is the an example:
```

Comment:
> `[lib.rs](http://lib.rs)`

Where does it link to?

Reply:
Doesn't intend to link to anywhere. Will remove it